### PR TITLE
upgrade and clean up eslint configuration

### PR DESCRIPTION
**Pull request by IgorSzyporyn**
## What I did

Updated `@storybook/eslint-config-storybook` to 3.1.0

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "root": true,
+  "extends": ["@storybook/eslint-config-storybook"],
+  "parserOptions": {
+    "project": ["tsconfig.json"],
+    "sourceType": "module"
```
Comment (ndelangen) [inline .eslintrc:6]: why would we switch to the non-js file format?

**Review by ndelangen — CHANGES_REQUESTED** [review]

**Comment by ndelangen** [comment]
@IgorSzyporyn the `3.1.0` version doesn't seem available on npm, could you check it out?

**Comment by IgorSzyporyn** [comment]
> @IgorSzyporyn the `3.1.0` version doesn't seem available on npm, could you check it out?

Just released 3.1.0 in `config-linter`

Have (this PR)[https://github.com/storybookjs/eslint-config-storybook/pull/11] up to get eslint-config-storybook up to speed, and then this will work

File: package.json [hunk 2]
```diff
@@ -154,9 +154,9 @@
     "@storybook/core-events": "workspace:*",
     "@storybook/csf-tools": "workspace:*",
     "@storybook/ember": "workspace:*",
-    "@storybook/eslint-config-storybook": "^2.4.0",
+    "@storybook/eslint-config-storybook": "3.1.0-beta.1",
```
Comment (IgorSzyporyn) [inline package.json:157]: change to 3.1.1

File: package.json [hunk 3]
```diff
@@ -154,9 +154,9 @@
...
     "@storybook/html": "workspace:*",
-    "@storybook/linter-config": "^2.5.0",
+    "@storybook/linter-config": "3.1.0-beta.1",
```
Comment (IgorSzyporyn) [inline package.json:159]: change to 3.1.1

**Comment by shilman** [comment]
@IgorSzyporyn what do you want to do about this one? can we close it for now?

**Comment by ndelangen** [comment]
related:
https://github.com/storybookjs/storybook/pull/15366

**Pull request closed without merging**